src/apps/config/leader.lua

```diff
@@ -83,21 +83,27 @@ function Leader:set_initial_configuration (configuration)
             '/', self.current_configuration)
 
    -- Iterate over followers starting the workers and queuing up actions.
-   for id, config in pairs(worker_app_graphs) do
-      local follower_pid = self:start_worker()
-      self.followers[id] = { pid=follower_pid, queue={}, graph=config}
-
-      -- Configure the follower.
-      local actions = self.support.compute_config_actions(
-	     app_graph.new(), self.followers[id].graph, {}, 'load')
-      self:enqueue_config_actions_for_follower(id, actions)
+   for id, worker_app_graph in pairs(worker_app_graphs) do
+      self:start_follower_for_graph(id, worker_app_graph)
    end
 end
 
 function Leader:start_worker()
    return worker.start("follower", self.worker_start_code)
 end
 
+function Leader:stop_worker(worker)
+   S.kill(worker.pid, 15)
```

Comment:
I think we should send a "stop_process" event to the worker, and have it shut down in an ordered fashion.  By doing an `engine.configure(config.new())`, the subprocess can result in a call to `stop()` on all apps, notably performing any cleanup needed by the intel_mp driver.  When we send this message to the follower, the follower will go into the 'stopping' state and we will poll on it periodically, trying to waitpid() it; once that succeeds, the follower can be removed.  Dunno, something like that.  This is for a followup though and I think I can get this one.

I think probably also we are missing a waitpid on the subprocess.